viskores/cont/CubicHermiteSpline.h

```diff
@@ -0,0 +1,105 @@
+//============================================================================
+//  The contents of this file are covered by the Viskores license. See
+//  LICENSE.txt for details.
+//
+//  By contributing to this file, all contributors agree to the Developer
+//  Certificate of Origin Version 1.1 (DCO 1.1) as stated in DCO.txt.
+//============================================================================
+
+//============================================================================
+//  Copyright (c) Kitware, Inc.
+//  All rights reserved.
+//  See LICENSE.txt for details.
+//
+//  This software is distributed WITHOUT ANY WARRANTY; without even
+//  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+//  PURPOSE.  See the above copyright notice for more information.
+//============================================================================
+#ifndef viskores_cont_CubicHermiteSpline_h
+#define viskores_cont_CubicHermiteSpline_h
+
+#include <viskores/Range.h>
+#include <viskores/Types.h>
+#include <viskores/cont/ArrayHandle.h>
+#include <viskores/cont/ExecutionObjectBase.h>
+#include <viskores/cont/UnknownArrayHandle.h>
+#include <viskores/cont/viskores_cont_export.h>
+#include <viskores/exec/CubicHermiteSpline.h>
+
+namespace viskores
+{
+namespace cont
+{
+
+class VISKORES_CONT_EXPORT CubicHermiteSpline : public viskores::cont::ExecutionObjectBase
+{
+public:
+  CubicHermiteSpline() = default;
+  virtual ~CubicHermiteSpline() = default;
+
+  VISKORES_CONT void SetData(const viskores::cont::ArrayHandle<viskores::Vec3f>& data)
+  {
+    this->Data = data;
+  }
+  VISKORES_CONT void SetData(const std::vector<viskores::Vec3f>& data,
+                             viskores::CopyFlag copy = viskores::CopyFlag::On)
+  {
+    this->Data.Allocate(0);
+    this->Data = viskores::cont::make_ArrayHandle(data, copy);
```

Comment:
This is not what I meant. The `Allocate` here has no real effect since the following assignment will clear out the old array. What I meant was that the `Knots` and `Tangents` arrays would be cleared out.

```suggestion
    this->Data = viskores::cont::make_ArrayHandle(data, copy);
    this->Knots.Allocate(0);
    this->Tangents.Allocate(0);
```

The reason for this is that if you set the data, use the data (so that the knots and tangents are computed) and then set the data again, the knots and tangents won't get updated unless you clear them out.